### PR TITLE
Add check to hide any old popup items when showing a new one

### LINKED ISSUE #5679: [Notebook] Delete Page PopUp Does Not Go Away
#### Summary
This has been copied from VIPEROMCT-160:
When clicked on the Shift Log for 20220729, the Delete Page popup appeared. When I subsequently clicked elsewhere it remained. Performed the following to try to clear it, which all failed:

Making a shift log entry
Switching to different days and different shifts
Navigating away to a different screen/layout
Refreshing the browser did clear it. And could not get the popup to stick.

Note from Khalid:
I was unable to get the delete popup to display while clicking a notebook page, but I was able to reproduce this while clicking the dropdown next to a notebook page twice. In this case the popup persists as described and nothing can be done to close it other than refreshing the page.

#### Steps to Reproduce
1. Go to a notebook and create a section and a page if none exist.
2. Click on the dropdown arrow next to a notebook page. The "Delete Page" dropdown should open.
3. Click on the dropdown arrow next to the same notebook page again (avoiding clicking on the "Delete Page" dropdown).
4. The dropdown will now persist despite trying to click elsewhere and there should be an error in the log.

#### Environment
* Open MCT Version: release/2.0.8
* OS: MacOS
* Browser: Chrome

#### Impact Check List
- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [X] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information
https://user-images.githubusercontent.com/2343076/185431129-182fa463-6433-4d9d-9014-17fb9691ea0d.mov

### PR DESCRIPTION
**Pull request by khalidadil**
Closes #5679

### Describe your changes:
Added a check to hide any old popup items when showing a new one

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@khalidadil would you mind adding a simple e2e test to capture this behavior in the regular notebook suite?

**Comment by khalidadil** [comment]
> @khalidadil would you mind adding a simple e2e test to capture this behavior in the regular notebook suite?

yup, it looks like the notebook tests are a bit light, so I was planning on writing an e2e test to make sure there's not a regression on this fix.

File: e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js [hunk 1]
```diff
@@ -67,10 +68,33 @@ test.describe('Default Notebook', () => {
...
+        //Navigate to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
+
+        // Create Notebook
+        await createDomainObjectWithDefaults(page, {
+            type: 'Notebook',
+            name: "Test Notebook"
+        });
+    });
+    test('Default sections are automatically named Unnamed Section with Unnamed Page', async ({ page }) => {
```
Comment (ozyx) [inline e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js:81]: These two tests could probably be combined into one. That would save us a tiny bit of runtime on the e2e tests.

You could even use `test.step()` to separate out the different test cases if you wanted to

File: e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js [hunk 2]
```diff
@@ -90,6 +114,30 @@ test.describe('Notebook section tests', () => {
...
+        await page.locator('.js-sidebar-pages .c-icon-button.icon-plus').click();
+
+        // Click on the 2nd page dropdown twice and then click on first page dropdown once
+        await page.locator('.c-popup-menu-button').nth(2).click();
+        await page.locator('.c-popup-menu-button').nth(2).click();
+        await page.locator('.c-popup-menu-button').nth(1).click();
+
+        // Check that there is only ever 1 Delete Page popup displayed
+        const numOfDeletePagePopups = await page.locator('div:has-text("Delete Page")').count();
+        expect(numOfDeletePagePopups).toBe(1);
```
Comment (ozyx) [inline e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js:139]: awesome!

**Review by ozyx — CHANGES_REQUESTED** [review]
Small suggestion on the e2e but looks great overall! Nice tests!

File: e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js (hunk 1 above)
Comment (khalidadil) [reply e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js]: makes sense! I just combined them into a single test

**Review by ozyx — APPROVED** [review]
Nicely done!

**Review by unlikelyzero — APPROVED** [review]
Amazing

**Comment by ozyx** [comment]
#2994

Might be fixed by this as well

**Comment by unlikelyzero** [comment]
@khalidadil there are merge conflicts. This is all that needs to be changed in the e2e test for now:
```
  //Test will need to be implemented after a refactor in #5713
    // eslint-disable-next-line playwright/no-skipped-test
    test.skip('Delete page popup is removed properly on clicking dropdown again', async ({ page }) => {
        test.info().annotations.push({
            type: 'issue',
            description: 'https://github.com/nasa/openmct/issues/5713'
        });
 ```

**Comment by ozyx** [comment]
> @khalidadil there are merge conflicts.

Went ahead and resolved those for you 🧚 ✨

**Comment by khalidadil** [comment]
> > @khalidadil there are merge conflicts.
> 
> Went ahead and resolved those for you 🧚 ✨

Thank you!! 🙏🏽

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**